backend/src/llm_client/base.py

```diff
@@ -0,0 +1,33 @@
+import abc
+from pydantic import BaseModel, Field
+from openai import OpenAI
+from enum import Enum
+
+class PromptPassingInterface(BaseModel):
+    system_content: str = Field(...)
+    user_content: str = Field(...)
+
+    @property
+    def to_dict(self):
+        value = [
+            {"role": "system", "content": f"{self.system_content}"},
+            {"role": "user", "content": f"{self.user_content}"},
+        ]
+        return value
```

Comment:
1. Noticing that you pass `...` to `Field`, I'd suggest removing `Field` since it's not required to define the attributes of BaseModel subclass.
2. The `to_dict()` method, with the `@property` decorator, can be accessed as an attribute. I'd recommend naming it as a noun for clarity
3. The `to_dict()` method returns a `list`, not a `dict`.
4. All of the instances `PromptPassingInterface` in you project are created by passing `system_content` and `user_content` directly (e.g., in src/llm_client/llm_client.py, lines 39 to 42):

```py
return self._generate(PromptPassingInterface(
             system_content="你是一個關鍵字提取機器人，用戶將會輸入一段文字，表示其希望看見的新聞內容，請提取出用戶希望看見的關鍵字，請截取最重要的關鍵字即可，避免出現「新聞」、「資訊」等混淆搜尋引擎的字詞。(僅須回答關鍵字，若有多個關鍵字，請以空格分隔)",
             user_content=prompt
         ))
```

If that's the case, inheriting from pydantic.BaseModel seems a bit redundant, and we can simply define it generally.

```suggestion
class PromptPassingInterface:
    def __init__(self, system_content: str, user_content: str):
        self.system_content = system_content
        self.user_content = user_content

    @property
    def llm_messages() -> list:
        ...
```